designs/0006-labels.md

```diff
@@ -0,0 +1,289 @@
+# Requirements and Labels
+
+**Author:** @matthchr
+
+**Last updated:** Oct 30, 2025
+
+**Status:** Proposed
+
+## Overview
+
+This document examines Karpenter and AKS managed labels and discusses what is needed to bring what we currently do more
+in line with AKS does.
+
+There are 2 main topics:
+* Label parity with AKS, setting labels that we don't currently set.
+* Support for scheduling simulation on labels we don't currently support, but should to enable easier migrations from cluster autoscaler to Karpenter.
+
+Today, Karpenter has the following categories of labels:
+
+1. Labels Karpenter understands and can schedule (Requirements)
+    * Can be set on Pod when not set on NodePool/NodeClaim
+    * Can be set on NodePool/NodeClaim
+    * May trigger behavior other than just setting a label (inform VM size, zone, etc)
+2. Labels written to the nodes but that Karpenter doesn't understand (can't schedule)
```

Comment:
Is it worth distinguishing between those passed to kubelet by the provider vs. those set via other mechanisms?

Reply:
> When you say "set by other mechanisms" you mean written by components other than Karpenter?

Yes. And Yes the distinction is already covered in the table, I was just wondering if is worth breaking down in the categorization here - but maybe not. As you said:

> Note that labels written by other parts of the system (daemonsets, operators, etc) generally fall into this category

Reply:
Yeah I think trying to keep this simple and just talk about the categories at a high-level that really matter for the purposes of "schedulable or not" is ideal. We can get into the nitty-gritty in the FAQ for the different sub-categories here if we want.